services/blockchain-indexer/shared/dataService/business/pos/unlocks.js

```diff
@@ -36,16 +38,24 @@ const getPosUnlocks = async params => {
 		{ address: params.address },
 	);
 
+	const { data: { height, genesis: { blockTime } } } = await getNetworkStatus();
+	const { timestamp } = await getBlockByHeight(height);
```

Comment:
If the block for the specified `height` isn't indexed, we could end up with incorrect information. Also, @vardan10 it'd be nice if used `lastBlockID` instead of `height` from the network status, so that we can utilize the cache within the connector.